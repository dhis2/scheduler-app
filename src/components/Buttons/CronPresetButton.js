import React, { useState } from 'react'
import { func } from 'prop-types'
import { Button } from '@dhis2/ui-core'
import i18n from '@dhis2/d2-i18n'
import { CronPresetModal } from '../Modal'

const CronPresetButton = ({ setCron }) => {
    const [showModal, setShowModal] = useState(false)

    return (
        <React.Fragment>
            <Button primary onClick={() => setShowModal(true)}>
                {i18n.t('Choose from preset times')}
            </Button>
            {showModal && (
                <CronPresetModal
                    hideModal={() => setShowModal(false)}
                    setCron={setCron}
                />
            )}
        </React.Fragment>
    )
}

CronPresetButton.propTypes = {
    setCron: func.isRequired,
}

export default CronPresetButton
