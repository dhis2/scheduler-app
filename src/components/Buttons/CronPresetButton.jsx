import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Button } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import { CronPresetModal } from '../Modal'

const CronPresetButton = ({ setCron, small }) => {
    const [showModal, setShowModal] = useState(false)

    return (
        <React.Fragment>
            <Button onClick={() => setShowModal(true)} small={small}>
                {i18n.t('Choose from preset times')}
            </Button>
            {showModal && (
                <CronPresetModal
                    hideModal={
                        /* istanbul ignore next */
                        () => setShowModal(false)
                    }
                    setCron={setCron}
                />
            )}
        </React.Fragment>
    )
}

CronPresetButton.defaultProps = {
    small: false,
}

const { func, bool } = PropTypes

CronPresetButton.propTypes = {
    setCron: func.isRequired,
    small: bool,
}

export default CronPresetButton
