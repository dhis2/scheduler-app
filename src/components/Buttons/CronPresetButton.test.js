import React from 'react'
import { shallow, mount } from 'enzyme'
import CronPresetButton from './CronPresetButton'

describe('<CronPresetButton>', () => {
    it('renders without errors', () => {
        shallow(<CronPresetButton setCron={() => {}} />)
    })

    it('renders without errors when small', () => {
        shallow(<CronPresetButton setCron={() => {}} small />)
    })

    it('shows the modal when button is clicked', () => {
        const wrapper = mount(<CronPresetButton setCron={() => {}} />)

        expect(wrapper.find('CronPresetModal')).toHaveLength(0)

        wrapper.find('button').simulate('click')

        expect(wrapper.find('CronPresetModal')).toHaveLength(1)
    })
})
