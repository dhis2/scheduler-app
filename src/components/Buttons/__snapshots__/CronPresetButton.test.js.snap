// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CronPresetButton> renders correctly 1`] = `
<Fragment>
  <Button
    dataTest="dhis2-uicore-button"
    onClick={[Function]}
    primary={true}
    type="button"
  >
    Choose from preset times
  </Button>
</Fragment>
`;
